docs/setup.md

```diff
@@ -0,0 +1,702 @@
+    ```
+    sudo -iu codecity
+    ```
+0.  Clone the [Code City
+    repo](https://github.com/NeilFraser/CodeCity).  (If you are a
+    project collaborator, see below for [instructions on how to use
+    SSH instad of HTTPS](#git-code-city-by-ssh-instead-of-https).)
+    ```
+    git clone https://github.com/google/CodeCity.git
+    ```
+0.  Install install required NPMs:
+    ```
+    (cd CodeCity/server && npm ci --only=prod)
+    (cd CodeCity/login && npm ci --only=prod)
+    ```
+0.  Exit from the `codecity` account.  We’re done with it for now, and
+    we need to be able to sudo, which that account (deliberately) does
+    not have permission to do.
+    ```
+    exit
+    ```
+
+### Configure NGINX
+
+On Debian, per-host `nginx` configuration files are stored in
+`/etc/nginx/sites-available` and enabled by symlinking them into
+`/etc/nginx-sites-enabled`.  There is a `default` config supplied by
+the `nginx` package, which should be disabled (unless you have already
+modified it to serve other virutal hosts).
+
+1.  Install the NGINX configuration file.  If you have a wildcard DNS
+    record and corresponding wildcard TLS certificate, use the
+    “subdomain” configuration:
+    ```
+    sudo cp ~codecity/CodeCity/etc/cc-subdomain.conf \
+            /etc/nginx/sites-available/codecity
+    ```
+    Otherwise, use the “onedomain” configuration:
+    ```
+    sudo cp ~codecity/CodeCity/etc/cc-onedomain.conf \
+            /etc/nginx/sites-available/codecity
+    ```
+0.  Edit /etc/nginx/sites-enabled to replace INSTANCENAME with the
+    name(s) of your instance.  (See comments for details.  You may use
+    another editor instead of nano if you wish!)
+    ```
+    sudo nano /etc/nginx/sites-available/codecity
+    ```
+0.  Enable the new configuration and reload (or restart) NGINX:
+    ```
+    sudo rm /etc/nginx/sites-enabled/default
+    sudo ln -s /etc/nginx/sites-available/codecity \
+               /etc/nginx/sites-enabled/codecity
+    sudo systemctl reload-or-restart nginx
+    ```
+
+### Create an API Key for OAuth
+
+The usual set-up for public Code City instances is to use [OAuth
+2.0](https://oauth.net/2/) via Google’s OAuth service for logins.
+This step will set up the necessary credentials to allow users to log
+in to your instance using their Google (Gmail) account.  You can skip
+this step if you intend to use a different login mechanism.
+
+This must be done after installing nginx and CodeCity and obtaining a
+TLS certificate because it depends on `logo-auth.png` being served by
+nginx.
+
+1.  Optionally replace `~codecity/CodeCity/static/logo-auth.png` with
+    a logo representing your instance or organisation.  It should be a
+    120x120px PNG image.
+0.  Make sure you can access your desired logo using your web browser.
+    If you are using a wildcard DNS configuration, it should be
+    accessible via a URL like
+    <code>static.<em>exmaple</em>.codecity.world/logo-auth.png</code>;
```

Comment:
'exmaple'
And two lines down.

Reply:
No, this is correct; I was referring to the code city instance run by former employees of the well known Waterloo software firm [Maple](https://www.maplesoft.com/).  ;-)